pallets/file-system/src/utils.rs

```diff
@@ -1400,6 +1400,11 @@ where
         // Update root of BSP.
         <T::Providers as shp_traits::MutateProvidersInterface>::update_root(bsp_id, new_root)?;
 
+        if new_root == <T::Providers as shp_traits::ReadProvidersInterface>::get_default_root() {
+            // We should remove the BSP from the dealer proof
+            <T::ProofDealer as shp_traits::ProofsDealerInterface>::stop_challenge_cycle(&bsp_id)?;
+        };
```

Comment:
This is the `confirm_storing` extrinsic, which means the only mutations this can have is trie add mutations (not remove), so the new root after apply delta should never be the default root (if it is, there's something wrong)

Reply:
I have removed it entirely. Not sure if I should still leave a check with an unreachable call...